kernel/src/client/sync/parquet.rs

```diff
@@ -1,12 +1,51 @@
+use std::fs::File;
+
+use arrow_schema::Schema as ArrowSchema;
+use parquet::arrow::arrow_reader::{ArrowReaderMetadata, ParquetRecordBatchReaderBuilder};
 use tracing::debug;
+use url::Url;
 
 use crate::{
-    schema::SchemaRef, DeltaResult, Expression, FileDataReadResultIterator, FileMeta,
-    ParquetHandler,
+    client::{
+        arrow_data::ArrowEngineData,
+        arrow_utils::{generate_mask, get_requested_indices, reorder_record_batch},
+    },
+    schema::SchemaRef,
+    DeltaResult, Error, Expression, FileDataReadResultIterator, FileMeta, ParquetHandler,
 };
 
 pub(crate) struct SyncParquetHandler;
 
+fn try_create_from_parquet(schema: SchemaRef, location: Url) -> DeltaResult<ArrowEngineData> {
+    let file = File::open(
+        location
+            .to_file_path()
+            .map_err(|_| Error::generic("can only read local files"))?,
```

Comment:
aside: Wow [Url::to_file_path](https://docs.rs/url/latest/url/struct.Url.html#method.to_file_path)... I've never seen `Err(())` before

Reply:
Hah, I know @hntd187 LOVES this error type :)

Reply:
Yeah, Nick has heard a good amount of my strong dislike of this error signature, I had hoped they would at least move to something like `Infallible` or `!` then it comes to stable, but yeah, still `()`

Reply:
I don't think `Err(())` is `Infallible`? It just means the error `Result` carries no information other than "I'm an error"?